vllm/worker/worker_base.py

```diff
@@ -81,3 +87,54 @@ def remove_lora(self, lora_id: int) -> bool:
 
     def list_loras(self) -> List[int]:
         raise ValueError(f"{type(self)} does not support LoRA")
+
+
+class WorkerWrapperBase:
```

Comment:
We should strive for composition over inheritance to limit the coupling of the implementations to the interface; otherwise we invite undesired coupling which makes some changes difficult.

Reply:
I don't get it. Can you elaborate on this?

Reply:
The thinking goes like this:
1. The purpose of an interface is to minimally couple different implementations.
2. Different implementations may have common logic, and this common logic can be put in the abstract class
3. It is good to avoid this because it violates (1); the coupling goes beyond the minimal interface definition
4. The alternative to inheritance is to have utility classes / functions and compose their usage within the different implementations. [example](https://chat.openai.com/share/6ff20436-023f-4ff6-9f2a-8925cb268d3d)

It is not a hard rule as it's impossible to completely decouple different components but a good design principle.

I'll still approve this PR without it; however the benefit is that each worker implementation is more cleanly separated from each other, making future changes easier.